src/cdk/drag-drop/directives/mixed-drop-list.spec.ts

```diff
@@ -0,0 +1,144 @@
+import {Component, QueryList, ViewChild, ViewChildren} from '@angular/core';
+import {fakeAsync, flush} from '@angular/core/testing';
+import {CdkDropList} from './drop-list';
+import {CdkDrag} from './drag';
+import {moveItemInArray} from '../drag-utils';
+import {CdkDragDrop} from '../drag-events';
+import {
+  ITEM_HEIGHT,
+  ITEM_WIDTH,
+  assertStartToEndSorting,
+  assertEndToStartSorting,
+  defineCommonDropListTests,
+} from './drop-list-shared.spec';
+import {createComponent, dragElementViaMouse} from './test-utils.spec';
+
+describe('mixed drop list', () => {
+  defineCommonDropListTests({
```

Comment:
Note that we don't have that many tests here, because we reuse the ~190 tests from the existing drop list. We add some more below to ensure that sorting works properly on mixed lists as well.